### PR TITLE
[net9.0] Update to preview7 versions

### PR DESCRIPTION
**Pull request by rmarinho**
### Description of Change

Move all packages to 9.0.100-preview.7

### CODE REVIEW

**Review by jonathanpeppers — COMMENTED** [review]
Are there some merge conflicts?

File: eng/Version.Details.xml [hunk 1]
```diff
@@ -1,139 +1,139 @@
...
+      <Sha>4e278fe17f69ea31fbdcbab74ac47ec6fa84914b</Sha>
     </Dependency>
-    <Dependency Name="Microsoft.Android.Sdk.Windows" Version="34.99.0-preview.6.340">
+    <Dependency Name="Microsoft.Android.Sdk.Windows" Version="34.99.0-preview.7.346">
       <Uri>https://github.com/dotnet/android</Uri>
-      <Sha>ef5fd216aa83867f13b8abdc0986d9373151a14e</Sha>
+      <Sha>06bb1dc6a292ef5618a3bb6ecca3ca869253ff2e</Sha>
     </Dependency>
-    <Dependency Name="Microsoft.MacCatalyst.Sdk.net9.0_17.2" Version="17.2.9714-net9-p6">
+    <Dependency Name="Microsoft.MacCatalyst.Sdk.net9.0_17.2" Version="17.2.9634-net9-p6">
```
Comment (rolfbjarne) [inline eng/Version.Details.xml:15]: This looks like a downgrade, where did you get this version from?

Comment (rmarinho) [reply eng/Version.Details.xml]: Seems thats the last one on the channel. But i reverted 

```
PS C:\repos\dotnet\maui> darc update-dependencies --channel ".NET 9.0.1xx SDK"
Looking up latest build of https://github.com/dotnet/sdk on .NET 9.0.1xx SDK
Looking up latest build of https://github.com/dotnet/android on .NET 9.0.1xx SDK
Looking up latest build of https://github.com/xamarin/xamarin-macios on .NET 9.0.1xx SDK
Looking up latest build of https://dev.azure.com/microsoft/ProjectReunion/_git/ProjectReunionInternal on .NET 9.0.1xx SDK
Looking up latest build of https://github.com/dotnet/aspnetcore on .NET 9.0.1xx SDK
Looking up latest build of https://github.com/dotnet/templating on .NET 9.0.1xx SDK
Looking up latest build of https://github.com/dotnet/xharness on .NET 9.0.1xx SDK
Looking up latest build of https://github.com/dotnet/arcade on .NET 9.0.1xx SDK
Updating 'Microsoft.MacCatalyst.Sdk.net9.0_17.2': '17.2.9714-net9-p6' => '17.2.9634-net9-p6' (from build '20240528.4' of 'https://github.com/xamarin/xamarin-macios')
Checking for coherency updates...
Local dependencies updated from channel '.NET 9.0.1xx SDK'.
```

File: eng/Version.Details.xml [hunk 2]
```diff
@@ -1,16 +1,16 @@
...
     </Dependency>
-    <Dependency Name="Microsoft.NETCore.App.Ref" Version="9.0.0-preview.6.24327.7" CoherentParentDependency="VS.Tools.Net.Core.SDK.Resolver">
+    <Dependency Name="Microsoft.NETCore.App.Ref" Version="9.0.0-preview.7.24357.2" CoherentParentDependency="VS.Tools.Net.Core.SDK.Resolver">
       <Uri>https://github.com/dotnet/runtime</Uri>
-      <Sha>573461dc897731c11d6166574b77a8c3aabba220</Sha>
+      <Sha>4e278fe17f69ea31fbdcbab74ac47ec6fa84914b</Sha>
     </Dependency>
-    <Dependency Name="Microsoft.Android.Sdk.Windows" Version="34.99.0-preview.6.340">
+    <Dependency Name="Microsoft.Android.Sdk.Windows" Version="34.99.0-preview.7.346">
       <Uri>https://github.com/dotnet/android</Uri>
```
Comment (jonathanpeppers) [inline eng/Version.Details.xml:12]: This is a slightly newer build than Preview 6, but the Android build is currently broken during the signing step.

We might not get a newer Android build than this, until this is merged:
* https://github.com/dotnet/android/pull/9091

**Review by jonathanpeppers — APPROVED** [review]

**Pull request merged**